### DISCUSSION TITLE
Will V support classes eventually?

### DISCUSSION THREAD

**Discussion by Jah-On** [vlang/v#12352] [Ideas]
Hi all,

I really like V so far and think it has a bright future. However, some of my projects require a more object oriented approach. While structs can mimick classes in a way, it doesn't offer protected zones and are awkward inheritance wise in my opinion. True class support would be a huge bonus.

Gracias.

**Comment by JalonSolov** [comment]
No, V will never have OOP-style classes.  However, it does support "protected" by the nature of it's module system.  Anything in a module that is not declared as public (with `pub` keyword) is not visible outside of the module, but it is visible to all the files in the module, which is similar to what the `protected` attribute in Java gives you.

**Comment by medvednikov** [comment]
V belongs to the new family of non OOP languages like Go and Rust.

**Comment by ghost** [comment]
I'm not sure what you exactly mean by the word "class", we can hide the privates by making them a module (confining it to a folder), although have to have folders and public getter/setter a lot since we don't have properties.

We can use delegation via embedding, although need to wrap the functions of the embedding struct by ourselves.

We can still write with an OOP like approach, although have to write a lot of boilerplate by hand.
Even if the syntax "class" is not introduced, the SOLID approach can still be used as Go does, so if V supports properties and the compiler automatically wraps functions of the embedded struct, whether it is OOP or not,  would be great to save time and effort.
